nntrainer/src/optimizer.cpp

```diff
@@ -58,6 +58,30 @@ int Optimizer::setOptParam(OptParam p) {
   return status;
 }
 
+int Optimizer::initialize(std::vector<Tensor> tensors, bool set_tensor) {
+  int status = ML_ERROR_NONE;
+
+  std::vector<Tensor>::iterator iter;
+
+  for (unsigned int i = 0; i < tensors.size(); i += 2) {
+    if (tensors[i].height() == 0 || tensors[i].width() == 0 ||
+        tensors[i].channel() == 0) {
+      ml_loge("Error: Tensor Dimension must be greater than 0");
+      return ML_ERROR_INVALID_PARAMETER;
+    }
+
+    if (tensors[i + 1].height() == 0 || tensors[i + 1].width() == 0 ||
+        tensors[i + 1].channel() == 0) {
+      ml_loge("Error: Tensor Dimension must be greater than 0");
+      return ML_ERROR_INVALID_PARAMETER;
+    }
```

Comment:
IMO. this is not necessary as tensorDim is using c++ style throw to check tensordim patched at https://github.com/nnstreamer/nntrainer/pull/234/commits/7a70a1fcc85b1f676171fe667852470c203feeac

Reply:
Right. fixed. Thanks.